### ISSUE TITLE
Can i make my chestmania only focusing super chests in LK

### ISSUE THREAD

**Issue by oband2** [kolton/d2bot-with-kolbot#1347]
I think my bot is wasting some time with 2 random non-superchests in LK, and also looking for weapon rack which i dont need.
Can i make my bot only focusing the 2-4 super chests in LK, if so, how?

ty in advance <3

**Comment by DarkHorseDre** [comment]
The following scripts list the chests to open, which are pretty much duplicated:
Follower.js
Attack.js
Misc.js

**Comment by oband2** [comment]
> The following scripts list the chests to open, which are pretty much duplicated:
> Follower.js
> Attack.js
> Misc.js

Thanks, however i dont see anything related to chest opening in Follower, and attack?
I seen some in misc, but i dont know what to change to get the adjustments that i want.
can u help with this?

**Comment by oband2** [comment]
i see this but, which is super chest, and which is normal?
openChests: function (range) {
		var unit,
			unitList = [],
			containers = ["chest", "chest3", "armorstand", "weaponrack"];

**Comment by Blitz28** [comment]
in common/attack.js lines 554, 725, 1012, 1013, 1024, and 1039 are chest opening commands.

in libs/bots/follower.js line 412 contains chest opening commands

**Comment by Blitz28** [comment]
chest3 i believe is super chest

**Comment by oband2** [comment]
> in common/attack.js lines 554, 725, 1012, 1013, 1024, and 1039 are chest opening commands.
> 
> in libs/bots/follower.js line 412 contains chest opening commands

Thanks, i will copy my lines in my scripts to show u why im confused:
Common/attack.js: - "blank" means nothing is written there
Line 554: "blank"
Line 725: 	var monster = getUnit(1);
Line 1012: currCount = this.getMonsterCount(grid[i].x, grid[i].y, range, monList);
Line: 1013: "blank"
Line 1024: "blank"
Line 1039:  count = 0,

Libs/bots/follower:
412:
containers = ["chest", "loose rock", "hidden stash", "loose boulder", "corpseonstick", "casket", "armorstand", "weaponrack", "barrel", "holeanim",
"roguecorpse", "ratnest", "corpse", "goo pile", "largeurn", "urn", "chest3", "jug", "skeleton", "guardcorpse", "sarcophagus",
"cocoon", "basket", "stash", "hollow log", "hungskeleton", "pillar", "skullpile", "skull pile", "jar3", "jar2", "jar1", "bonechest", "woodchestl",
"woodchestr", "barrel wilderness", "burialchestr", "burialchestl", "explodingchest", "chestl", "chestr", "icecavejar1", "icecavejar2",
"icecavejar3", "icecavejar4", "deadperson", "deadperson2", "evilurn", "tomb1l", "tomb3l", "tomb2", "tomb3", "object2", "groundtomb", "groundtombl"

However i dont see why line 412 is related, since it doesnt open hollow logs, corpse, dead person or whatever other things there is to open in Lower kurast is mentioned in this line.
The only things its opening is 2 small chests, all super chests, all weapon racks, all armor racks.

**Comment by DarkHorseDre** [comment]
I grep'd the files for completeness, but you don't need to worry about scripts you're not using:
'follower.js' is the script you use to make a leecher follow a **manually** playing leecher - you're not doing that. The follower list is only used by followers, not your leader/solo mfer

attack.js and Misc.js use the same list (but duplicate instead of using a common list) and you can expand that list to what you see in attack.js under the alternative config check:
'if (Config.OpenChests === 2)'

When you set openchests in your config you haven't selected 2, so the expanded list isn't used but you can set it, or add desired chests to your standard list ("chest", "chest3", "armorstand", "weaponrack")

**Comment by oband2** [comment]
I dont a have line called anything with (Config.OpenChests === 2) in my attack.js, do u mean misc?

And i should edit that into something like "(Config.OpenChests === "chest3")
so it should work?

**Comment by DarkHorseDre** [comment]
No, I said `When you set openchests in your config you haven't selected 2`

**Comment by oband2** [comment]
I can set true or false in my config?
where should i select 2?

Config.OpenChests = true; // Open chests. Controls key buying.

**Comment by DarkHorseDre** [comment]
bro, what I have explained is that the code will accept a 2 if you set it, which you havent, thus it uses the other values.
if you set it to 2 ensure you understand the implications. else, just fix the issue you have, which is to open the specific chests in LK.

**Comment by oband2** [comment]
Ok, so i do like this:
Config.OpenChests = 2; // Open chests. Controls key buying.

e.g 
if (Config.OpenChests === 2) {
			containers = [
				"chest3"

**Comment by DarkHorseDre** [comment]
yes, the only thing you need to do to use the full list is to set the option to 2.

**Comment by DarkHorseDre** [comment]
I notice you have created many issues - you should search existing ones first: https://github.com/kolton/d2bot-with-kolbot/issues/1239

**Comment by oband2** [comment]
Okay thanks, i just didnt understand it that easy the way u explained it, but thanks will test asap

**Comment by Karyoplasma** [comment]
I created a new script specifically for LK runs. I called it LKFarm.js and dropped it into the kolbot/libs/bot folder. In your config, make sure Scripts.UserAddon, Config.OpenChests and ALL scripts are set to false. I also recommend setting cainID and fieldID to true, but that's your choice. Then I added a line Scripts.LKFarm = true (you can add it anywhere, I placed it above ChestMania) and watched my Sorc farm. 

Here is the pastebin of my script: https://pastebin.com/aA45pdKp

 I annotated so you can learn what is going on.

**Comment by DarkHorseDre** [comment]
Hey @Karyoplasma thanks for sharing!  Works nicely!

I'm testing in SP and its dropping lots!
